cluster-operator/src/main/java/io/strimzi/operator/cluster/model/CruiseControl.java

```diff
@@ -0,0 +1,497 @@
+                cruiseControl.templateDeploymentAnnotations = template.getDeployment().getMetadata().getAnnotations();
+            }
+
+            if (template.getApiService() != null && template.getApiService().getMetadata() != null) {
+                cruiseControl.templateServiceLabels = template.getApiService().getMetadata().getLabels();
+                cruiseControl.templateServiceAnnotations = template.getApiService().getMetadata().getAnnotations();
+            }
+
+            if (template.getCruiseControlContainer() != null && template.getCruiseControlContainer().getEnv() != null) {
+                cruiseControl.templateCruiseControlContainerEnvVars = template.getCruiseControlContainer().getEnv();
+            }
+
+            if (template.getTlsSidecarContainer() != null && template.getTlsSidecarContainer().getEnv() != null) {
+                cruiseControl.templateTlsSidecarContainerEnvVars = template.getTlsSidecarContainer().getEnv();
+            }
+
+            ModelUtils.parsePodDisruptionBudgetTemplate(cruiseControl, template.getPodDisruptionBudget());
+        }
+        return cruiseControl;
+    }
+
+    static List<Toleration> tolerations(CruiseControlSpec spec) {
+        if (spec.getTemplate() != null
+                && spec.getTemplate().getPod() != null
+                && spec.getTemplate().getPod().getTolerations() != null) {
+            return spec.getTemplate().getPod().getTolerations();
+        }
+        return null;
+    }
+
+    static Affinity affinity(CruiseControlSpec spec) {
+        if (spec.getTemplate() != null
+                && spec.getTemplate().getPod() != null
+                && spec.getTemplate().getPod().getAffinity() != null) {
+            return spec.getTemplate().getPod().getAffinity();
+        }
+        return null;
+    }
+
+    public static String cruiseControlName(String cluster) {
+        return CruiseControlResources.deploymentName(cluster);
+    }
+
+    public static String cruiseControlServiceName(String cluster) {
+        return CruiseControlResources.serviceName(cluster);
+    }
+
+    public Service generateService() {
+        if (!isDeployed()) {
+            return null;
+        }
+
+        List<ServicePort> ports = new ArrayList<>(1);
+        ports.add(createServicePort(REST_API_PORT_NAME, REST_API_PORT, REST_API_PORT, "TCP"));
+
+        return createService("ClusterIP", ports, templateServiceAnnotations);
+    }
+
+    protected List<ContainerPort> getContainerPortList() {
+        List<ContainerPort> portList = new ArrayList<>(1);
+
+        portList.add(createContainerPort(REST_API_PORT_NAME, REST_API_PORT, "TCP"));
+
+        if (isMetricsEnabled) {
+            portList.add(createContainerPort(METRICS_PORT_NAME, METRICS_PORT, "TCP"));
+        }
+
+        return portList;
+    }
+
+    protected List<Volume> getVolumes(boolean isOpenShift) {
+        List<Volume> volumeList = new ArrayList<>(1);
```

Comment:
If you insist on doing it like this then get the right number of items:

```suggestion
        List<Volume> volumeList = new ArrayList<>(3);
```

But even better would be to use 
```java
return Arrays.asList(createSecretVolume(...),
                     createSecretVolume(...),
                     createConfigMapVolume(...));
```